FirebaseABTesting/Sources/FIRExperimentController.m

```diff
@@ -34,19 +35,19 @@
 #define STR_EXPAND(x) #x
 
 /// Default experiment overflow policy.
-const ABTExperimentPayload_ExperimentOverflowPolicy FIRDefaultExperimentOverflowPolicy =
-    ABTExperimentPayload_ExperimentOverflowPolicy_DiscardOldest;
+const ABTExperimentPayloadExperimentOverflowPolicy FIRDefaultExperimentOverflowPolicy =
+    ABTExperimentPayloadExperimentOverflowPolicyDiscardOldest;
 
 /// Deserialize the experiment payloads.
 ABTExperimentPayload *ABTDeserializeExperimentPayload(NSData *payload) {
+  // Verify that we have a JSON object.
   NSError *error;
-  ABTExperimentPayload *experimentPayload = [ABTExperimentPayload parseFromData:payload
-                                                                          error:&error];
-  if (error) {
+  id JSONObject = [NSJSONSerialization JSONObjectWithData:payload options:kNilOptions error:&error];
+  if (JSONObject == nil) {
```

Comment:
When there's no experiment running, is it possible that JSON object is just nil without error?

Reply:
Since this method gets called from an enumeration, the payload data should never be nil.  Only way this serialization should fail is if it's a non-JSON object.